### DISCUSSION TITLE
Announcement: ZeppOS Tasks — ToDo with Google Tasks sync

### DISCUSSION THREAD

**Discussion by melianmiko** [zepp-health/zeppos-forum#193] [General]
**Features:**
- View tasks, mark them as completed
- Create new tasks with on-screen keyboard
- Offline mode: cache task list and changes for use without internet
- Sync with Google Tasks )optional, also could work without)

**Supported devices:** all with ZeppOS 1.0 and newer (including Mi Band 7).

Available in Zepp App Store, google moderation completed too. If you're Mi Band 7 user, you can install application from [my website](https://melianmiko.ru/en/zepp/tasks/).

Source code [available](https://github.com/melianmiko/ZeppOS-Tasks). Pull requests with translations to other languages are welcome. More keyboard layouts also would be added later.

![28338338](https://github.com/zepp-health/zeppos-forum/assets/67851274/3185a6a3-9800-4c61-867e-e40a91a82dd5)
![28338337](https://github.com/zepp-health/zeppos-forum/assets/67851274/ed143834-a232-4672-af72-03111fc48a6f)
![28338334](https://github.com/zepp-health/zeppos-forum/assets/67851274/1327fab7-8e88-49c0-88aa-d8e9e5ac0674)
![28338335](https://github.com/zepp-health/zeppos-forum/assets/67851274/63165122-2a38-4aae-8f24-9c65a6b297c8)

**Comment by MyoungXUE** [comment]
@melianmiko 
Hello, this is a very good project with a high degree of completion and a very standard project. Could we include it in https://github.com/zepp-health/awesome-zeppos.

Also let me tell you that we are actively promoting the approval of the Zepp app store.

**Reply by melianmiko** [reply]
Yes, of couse you can add it to your awesome list.

**Reply by melianmiko** [reply]
@MyoungXUE 
Sorry for annoying, but I don't know who or where ask about that.

Why my application got this warning in Zepp App Store? They didn't have any paid features, application is totally free. The only thing that they contain is a link to make donations, which didnt give any additional features and is absolutely optional.

![Screenshot_20230717-113803_Zepp_1.jpg](https://github.com/zepp-health/zeppos-forum/assets/67851274/9aafe556-149d-41d1-820b-f3f9449fa222)

**Reply by MyoungXUE** [reply]
@melianmiko Hello, I'm sorry I just saw this message, I will feedback this issue with the relevant colleagues, please be patient!

**Reply by MyoungXUE** [reply]
@melianmiko

Our system mistakenly flagged this app as a paid app while performing a batch task, resulting in this line of alert text, which has now been corrected.

We apologize again for the bad experience.

**Reply by melianmiko** [reply]
@MyoungXUE, no problem. Thank you for fast response.

**Comment by Cotrox** [comment]
When i visit login url he says 'invalid indent'

**Reply by melianmiko** [reply]
Hi,
Press "Cancel" on settings page and try to log in again. Looks like it's server bug, I'll check them later.

**Comment by Cotrox** [comment]
It is ok thanks!

Giuseppe A. Cotroneo
________________________________
Da: MelianMiko ***@***.***>
Inviato: Tuesday, July 4, 2023 11:05:53 AM
A: zepp-health/zeppos-forum ***@***.***>
Cc: Giuseppe Antonino Cotroneo ***@***.***>; Comment ***@***.***>
Oggetto: Re: [zepp-health/zeppos-forum] Announcement: ZeppOS Tasks — ToDo with Google Tasks sync (Discussion #193)


Hi,
Press "Cancel" on settings page and try to log in again. Looks like it's server bug, I'll check them later.

**Comment by Cotrox** [comment]
Can you add Microsoft To Do version?

Giuseppe A. Cotroneo
________________________________
Da: Giuseppe Antonino Cotroneo ***@***.***>
Inviato: Tuesday, July 4, 2023 11:06:46 AM
A: zepp-health/zeppos-forum ***@***.***>; zepp-health/zeppos-forum ***@***.***>
Cc: Comment ***@***.***>
Oggetto: Re: [zepp-health/zeppos-forum] Announcement: ZeppOS Tasks — ToDo with Google Tasks sync (Discussion #193)

It is ok thanks!

Giuseppe A. Cotroneo
________________________________
Da: MelianMiko ***@***.***>
Inviato: Tuesday, July 4, 2023 11:05:53 AM
A: zepp-health/zeppos-forum ***@***.***>
Cc: Giuseppe Antonino Cotroneo ***@***.***>; Comment ***@***.***>
Oggetto: Re: [zepp-health/zeppos-forum] Announcement: ZeppOS Tasks — ToDo with Google Tasks sync (Discussion #193)


Hi,
Press "Cancel" on settings page and try to log in again. Looks like it's server bug, I'll check them later.

**Reply by melianmiko** [reply]
Maybe, later...

**Reply by melianmiko** [reply]
@Cotrox, sync with Ms ToDo will be available in next update.

**Reply by Cotrox** [reply]
Wow it will be amazing!

Have you a release date?

**Reply by melianmiko** [reply]
@Cotrox, it's already released and only waiting for Zepp moderation. 

If you want get it now, you can install fresh version from [my website](https://mmk.pw/en/zepp/tasks/), or wait for a few days and update from Zepp App Store.

**Reply by Cotrox** [reply]
Thank you! I installed it but i read 'can not read property timeout of undefined' 🥺

**Reply by Cotrox** [reply]
After 'wipe all data' it worked. But second time i opened the app i read disconnected or unknown action.

**Reply by melianmiko** [reply]
Tried on my watch, all work as expected. Maybe app didn't fully updated, try to uninstall Tasks by trash icon in developer mode list and install them again. 

Or maybe your watch lose connection to phone...

**Reply by Cotrox** [reply]
Sometimes it works but sometimes not 🥺

**Comment by Cotrox** [comment]
😫😫😫 ahah no worries I know. It is not so easy probably 😌

Giuseppe A. Cotroneo
________________________________
Da: MelianMiko ***@***.***>
Inviato: Tuesday, July 4, 2023 1:34:14 PM
A: zepp-health/zeppos-forum ***@***.***>
Cc: Giuseppe Antonino Cotroneo ***@***.***>; Comment ***@***.***>
Oggetto: Re: [zepp-health/zeppos-forum] Announcement: ZeppOS Tasks — ToDo with Google Tasks sync (Discussion #193)


Maybe, later...

**Reply by htoooth** [reply]
I want to write a Microsoft To Do version for Zepp OS, but I don't have enough time recently.

**Comment by Cotrox** [comment]
I can understand, i hope you will find more time
Asap ahah

Giuseppe A. Cotroneo
________________________________
Da: htoooth ***@***.***>
Inviato: Wednesday, July 19, 2023 10:10:11 AM
A: zepp-health/zeppos-forum ***@***.***>
Cc: Giuseppe Antonino Cotroneo ***@***.***>; Comment ***@***.***>
Oggetto: Re: [zepp-health/zeppos-forum] Announcement: ZeppOS Tasks — ToDo with Google Tasks sync (Discussion #193)


I want to write a Microsoft To Do version for Zepp OS, but I don't have enough time recently.

**Comment by Cotrox** [comment]
Thank youuuu!

Giuseppe A. Cotroneo
________________________________
Da: MelianMiko ***@***.***>
Inviato: Friday, January 12, 2024 11:36:14 AM
A: zepp-health/zeppos-forum ***@***.***>
Cc: Giuseppe Antonino Cotroneo ***@***.***>; Mention ***@***.***>
Oggetto: Re: [zepp-health/zeppos-forum] Announcement: ZeppOS Tasks — ToDo with Google Tasks sync (Discussion #193)


@Cotrox<https://github.com/Cotrox>, it's already released and only waiting for Zepp moderation.

If you want get it now, you can install fresh version from my website<https://mmk.pw/en/zepp/tasks/>, or wait for a few days and update from Zepp App Store.
